pkg/tfbridge/info/validate.go

```diff
@@ -107,48 +106,56 @@ func (c *infoCheck) checkProperty(path walk.SchemaPath, tfs shim.Schema, ps *Sch
 		return
 	}
 
-	// s.Elem() case 2
-	//
-	// `path` represents a singly-nested Terraform block.
-	//
-	// Conceptually `path` and `path.Elem` represent the same object.
-	//
-	// To prevent confusion, users are barred from specifying information on
-	// the associated Elem. All information should be specified directly on
-	// this SchemaInfo.
-	if obj, ok := util.CastToTypeObject(tfs); ok {
-		if ps.Elem != nil {
-			c.error(path, errElemForObject)
-		}
-
-		if ps.MaxItemsOne != nil {
-			c.error(path.Element(), errCannotSetMaxItemsOne)
-		}
-
-		// Now check sub-fields
-		c.checkFields(path, obj, ps.Fields)
-
-		return
-	}
-
-	if len(ps.Fields) > 0 {
-		c.error(path, errCannotSpecifyFieldsOnListOrSet)
-	}
-
 	switch elem := tfs.Elem().(type) {
 
 	// s.Elem() case 1
 	//
 	// There is a simple element type here, so we just recursively validate that.
 	case shim.Schema:
+		if len(ps.Fields) > 0 {
+			c.error(path, errCannotSpecifyFieldsOnListOrSet)
```

Comment:
Can we rename the variable to `CannotSpeciftFieldsOnCollection` - it has kept the old name?